node/node_queries_task.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+from typing import Any
+
+import aiohttp
+
+from common.batch import stateful_batch_to_batch_record
+from common.db import zdb
+from common.batch import BatchRecord
+from common.logger import zlogger
+from config import zconfig
+
+
+def find_all_nodes_last_finalized_batch_record(app_name: str) -> BatchRecord:
+    """
+    Synchronous wrapper to find the last finalized batch record.
+    For async contexts, use find_all_nodes_last_finalized_batch_record_async instead.
+    """
+    try:
+        return asyncio.run(_find_all_nodes_last_finalized_batch_record_core(app_name))
+    except Exception as e:
+        zlogger.error(f"Critical error in find_all_nodes_last_finalized_batch_record: {str(e)}")
+        # Return local record as ultimate fallback
+        return zdb.get_last_operational_batch_record_or_empty(
+            app_name=app_name,
+            state="finalized"
+        )
+
+
+async def find_all_nodes_last_finalized_batch_record_async(app_name: str) -> BatchRecord:
+    """
+    Async wrapper to find the last finalized batch record of network.
+    Can be called from async contexts.
+    """
+    try:
+        # Get the result asynchronously
+        return await _find_all_nodes_last_finalized_batch_record_core(app_name)
+    except Exception as e:
+        zlogger.error(f"Critical error in find_all_nodes_last_finalized_batch_record: {str(e)}")
+        # Return local record as ultimate fallback
+        return zdb.get_last_operational_batch_record_or_empty(
+            app_name=app_name,
+            state="finalized"
+        )
+
+
+async def _fetch_node_last_finalized_batch_record_or_empty(
+        session: aiohttp.ClientSession,
+        node: dict[str, Any],
+        app_name: str
+) -> BatchRecord:
+    """Fetch last finalized batch from a single node asynchronously."""
+    url = f'{node["socket"]}/node/{app_name}/batches/finalized/last'
```

Comment:
Let's add a new route to nodes to get the highest finalized for all apps and query that route here instead of sending different requests to each node for different apps.